apps/core/src/domain/actions/formatDateAction.spec.ts

```diff
@@ -9,31 +9,121 @@ describe('formatDateAction', () => {
+            expect(
+                (
+                    await action(
+                        [{value: testingDate}],
+                        {Dynamic},
+                        {
+                            ...ctx,
+                            lang: 'ar-EG'
+                        }
+                    )
+                ).values[0].value
+            ).toBe('٢٨ فبراير ٣٧ في ١١ م');
+        });
+
+        test.skip.each(['{', '{withoutDoubleQuote: true}', '', '{"params1": "long", "param2": "too many coma",}'])(
+            'auto should print default on invalid json format: `%s`',
+            async Dynamic => {
+                const result = await action(
+                    [{value: testingDate}],
+                    {Dynamic},
+                    {
+                        ...ctx,
+                        lang: 'en-EN'
+                    }
+                );
+                expect(result.values[0].value).toBe('2/28/2037, 11:42:00 PM');
+                expect(result.errors[0]).toEqual({
+                    attributeValue: {value: Dynamic},
+                    errorType: 'FORMAT_ERROR',
+                    message: 'Options of FormatDateAction are invalid JSON. Use `{}` empty option instead.'
+                });
+            }
+        );
+    });
+
+    test('Imposing format', async () => {
         expect(
-            (await action([{value: 2119477320}], {auto: 'false', format: 'D/M/YY HH:mm'}, ctx)).values[0].value
-        ).toBeTruthy();
-        expect((await action([{value: 2119477320}], {auto: 'false'}, ctx)).values[0].value).toBeTruthy();
-        expect((await action([{value: 'aaaa'}], {auto: 'false'}, ctx)).values[0].value).toBe('');
-        expect((await action([{value: null}], {auto: 'false'}, ctx)).values[0].value).toBe(null);
+            (
+                await action(
+                    [{value: testingDate}],
+                    {
+                        'Imposing format': 'D/MMMM-YY HH:mm'
+                    },
+                    ctx
+                )
+            ).values[0].value
+        ).toBe('28/February-37 23:42');
     });
 
-    test('auto', async () => {
-        expect((await action([{value: 2119477320}], {auto: 'true'}, {...ctx, lang: 'ko-KR'})).values[0].value).toBe(
-            '2037. 2. 28. 오후 11:42:00'
-        );
-        expect((await action([{value: 2119477320}], {auto: 'true'}, {...ctx, lang: 'fr-FR'})).values[0].value).toBe(
-            '28/02/2037 23:42:00'
-        );
-        expect((await action([{value: 2119477320}], {auto: 'true'}, {...ctx, lang: 'ar-EG'})).values[0].value).toBe(
-            '٢٨‏/٢‏/٢٠٣٧، ١١:٤٢:٠٠ م'
-        );
+    describe('edge cases', () => {
+        test('should display default localString on neither params provided', async () => {
+            expect((await action([{value: testingDate}], {}, ctx)).values[0].value).toBe(
+                (await action([{value: testingDate}], {Dynamic: '{}'}, ctx)).values[0].value
```

Comment:
**question:** Je ne comprends pas ce que fait ce test.

Reply:
le test s'appelle `should display default localString on neither params provided`, donc c'est la même chose que de ne pas mettre le paramètre que de le mettre à un objet `{}`, je ne savais pas trop comment l'écrire